### PR TITLE
use go-licenses to update licenses

### PR DESCRIPTION
**Pull request by chizhg**
**What this PR does, why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes to reviewers**:

**User-visible changes in this PR**:

/cc @chaodaiG @n3wscott 

/hold
Let me test if first.

### CODE REVIEW

File: scripts/library.sh [hunk 1]
```diff
@@ -500,12 +500,17 @@ function run_go_tool() {
...
 # Parameters: $1 - output file, relative to repo root dir.
-#             $2...$n - directories and files to inspect.
+#             $2 - directory to inspect.
 function update_licenses() {
   cd "${REPO_ROOT_DIR}" || return 1
   local dst=$1
+  local dir=$2
   shift
-  run_go_tool knative.dev/test-infra/tools/dep-collector dep-collector $@ > ./${dst}
+  go-licenses save "${dir}" --save_path="${dst}" --force
```
Comment (n3wscott) [inline scripts/library.sh:509]: I think you need to install this too

File: scripts/library.sh [hunk 2]
```diff
@@ -500,12 +500,17 @@ function run_go_tool() {
 
 # Run dep-collector to update licenses.
```
Comment (n3wscott) [inline scripts/library.sh:501]: run go-licenses*

File: scripts/library.sh [hunk 3]
```diff
@@ -500,12 +500,17 @@ function run_go_tool() {
...
   shift
-  run_go_tool knative.dev/test-infra/tools/dep-collector dep-collector $@ > ./${dst}
+  go-licenses save "${dir}" --save_path="${dst}" --force
+  # Hack to make sure directories retain write permissions after save. This
+  # can happen if the directory being copied is a Go module.
+  # See https://github.com/google/go-licenses/issues/11
+  chmod -R +w "${dst}"
 }
 
 # Run go-licenses to check for forbidden liceses.
```
Comment (n3wscott) [inline scripts/library.sh:516]: update this function too. and run go-licenses*

Comment (n3wscott) [reply scripts/library.sh:516]: see https://github.com/tektoncd/plumbing/blob/becdc43e0be0c57d85e88f2714a0a1ef670cc72f/scripts/library.sh#L344-L346

**Comment by n3wscott** [comment]
lgtm!!!

File: scripts/library.sh (hunk 2 above)
Comment (chizhg) [reply scripts/library.sh]: Fixed.

File: scripts/library.sh (hunk 1 above)
Comment (chizhg) [reply scripts/library.sh]: Added `run_go_tool`

File: scripts/library.sh (hunk 3 above)
Comment (chizhg) [reply scripts/library.sh]: This one has already been replaced.

**Pull request merged**